File: apps/monolith/src/main/java/com/example/iter/admin/stats/controller/AdminStatsApiController.java
package com.example.iter.admin.stats.controller;

import com.example.iter.admin.stats.controller.spec.AdminStatsApiSpec;
import com.example.iter.admin.stats.dto.response.AdminStatsResponse;
import com.example.iter.admin.stats.service.AdminStatsService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@Validated
@RestController
@RequestMapping("/api/v1/admin/stats")
@RequiredArgsConstructor
@PreAuthorize("hasRole('ADMIN')")
public class AdminStatsApiController implements AdminStatsApiSpec {

    private final AdminStatsService adminStatsService;

    // 관리자 대시보드 상단에 표시할 전체 회원·장비·접수된 신고·결제 건수를 조회합니다.
    @Override
    @GetMapping
    public ResponseEntity<AdminStatsResponse> getStats() {
        return ResponseEntity.ok(adminStatsService.getStats());
    }
}

File: apps/monolith/src/main/kotlin/com/example/iter/admin/stats/controller/AdminStatsApiController.kt
package com.example.iter.admin.stats.controller

import com.example.iter.admin.stats.controller.spec.AdminStatsApiSpec
import com.example.iter.admin.stats.dto.response.AdminStatsResponse
import com.example.iter.admin.stats.service.AdminStatsService
import org.springframework.http.ResponseEntity
import org.springframework.security.access.prepost.PreAuthorize
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/v1/admin/stats")
@PreAuthorize("hasRole('ADMIN')")
class AdminStatsApiController(private val adminStatsService: AdminStatsService) : AdminStatsApiSpec {
    @GetMapping
    override fun getStats(): ResponseEntity<AdminStatsResponse> =
        ResponseEntity.ok(adminStatsService.getStats())
}
